docs/UserGuide.md

```diff
@@ -32,103 +33,138 @@
-Format: `add NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]...` 
+#### Adding an item: `add`
+Adds an item into the task manager<br>
+
+*With deadline*
+
+Format: `add NAME by DUE_DATE DUE_TIME [repeat every RECURRING_INTERVAL]`
+>Notice the `by` keyword? We use `by` to denote a deadline.
+
+Examples:
+* `add Do project proposal by 5pm tomorrow`
+* `add Buy coffee for boss by 7am repeat every day`
+
+*Without deadline*
+
+Format: `add NAME [rank PRIORITY]`
+>No deadlines? No worries. DearJim allows you to rank your priorities. Assign rank of 1 to 10, with 1 being the most important task. If you did not specify the priority, it will be assigned to a default value of 5.
+
+Example:
+* `add Buy coffee powder rank 1`
+
+*With time interval*
+
+`add NAME at DATE START_TIME to [DATE] END_TIME [repeat every RECURRING_INTERVAL]` 
+> For events, meetings, we use `at`, `to` to specify the date/time interval.
+
+Example: 
+* `add Company meeting tonight at 7pm to 9pm`
+
+*Tentative event*
+
+Format: `add NAME (at DATE START_TIME to [DATE] END TIME...)` 
+> Sometimes, we have an unconfirmed event slot. For example, the meeting your boss wants to hold might be on 2-3pm tomorrow, or the day after. 
+> Group the list of dates inside `()` and separate each date with `,`
+
+Example: 
+* `add Company meeting (at 22 Sep 2pm to 3pm, at 23 Sep 2pm to 3pm)`
+
+#### Editing an item: `edit`
+Edits any item in the task manager<br>
+
+*Any type except tentative*
+
+Format: Replace 	`add` with `edit INDEX`
+>Tip: We make the edit commands similar to add commands, so that you only have to learn one of them! Simply replace `add` with `edit INDEX` to do an edit.
+
+Examples:
+* `edit 10 Company meeting tomorrow morning at 7am to 9am`
+* `edit 3 Buy coffee for boss by 7am repeat every 2 days`
 
-> Persons can have any number of tags (including 0)
+*Tentative events*
 
-Examples: 
-* `add John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01`
-* `add Betsy Crowe p/1234567 e/betsycrowe@gmail.com a/Newgate Prison t/criminal t/friend`
+Format: Replace `add` with  `edit INDEX (new OR SUB-INDEX)`
+>Note: If you want to confirm the time slot, use the `confirm` command instead.
 
-#### Listing all persons : `list`
-Shows a list of all persons in the address book.<br>
+Example:
+> <img src="images/edit_tentative_ui.png" width="600">
+
+* `edit 12 1 (at 24 Sep 8pm to 9pm) 3 (at 25 Sep 8pm to 9pm)`
+
+
+#### Confirming a tentative event: `confirm`
+Confirm a tentative event in your task manager. Must have added a tentative event first. Refer to `add` command.<br>
+
+*Tentative event*
+
+Format: `confirm INDEX SUBINDEX`
+
+Example:
+> <img src="images/confirm_ui.png" width="600">
+
+* `confirm 3 1`
+
+
+#### Deleting an item: `delete`
+Delete an item in your task manager This process is reversible with the `undo` command.<br>
+Format:
+`delete INDEX` `delete all`
+
+#### Undoing a command: `undo`
+Format: `undo`
+
+
+#### Redoing a command: `redo`
```

Comment:
Redo is just going forwards through the "undoed" commands. Must have undoed to be able to redo